src/csharp/OnnxRuntimeGenAIChatClient.cs

```diff
@@ -267,15 +269,41 @@ private bool IsStop(string token, ChatOptions? options) =>
         _options?.StopSequences?.Contains(token) is true;
 
     /// <summary>Formats messages into a prompt using a default format.</summary>
-    private static string FormatPromptDefault(IEnumerable<ChatMessage> messages, ChatOptions? options)
+    private string FormatPromptDefault(IEnumerable<ChatMessage> messages, ChatOptions? options)
     {
-        StringBuilder sb = new();
+        SerializableMessage m = new();
+
+        StringBuilder prompt = new();
+        string separator = "";
+        prompt.Append('[');
         foreach (var message in messages)
         {
-            sb.Append(message).AppendLine();
+            if (message.Text is string text)
+            {
+                prompt.Append(separator);
+                separator = ",";
+
+                m.Role = message.Role.Value;
+                m.Content = text;
+                prompt.Append(JsonSerializer.Serialize(m));
```

Comment:
Shouldn't this specify the `JsonTypeInfo`?
```
prompt.Append(JsonSerializer.Serialize(m, OnnxJsonContext.Default.SerializableMessage));
```

Reply:
It should, yes, not sure how I managed to drop that.

I'll also submit a separate PR to enable the NativeAOT analyzers, which would have also caught this.